float32 cur_x
float32 cur_y
float32 dest_x
float32 dest_y

